src/Microsoft.TemplateEngine.Abstractions/IGenerator.cs

```diff
@@ -57,6 +58,29 @@ public interface IGenerator : IIdentifiedComponent
         /// <returns><see cref="IParameterSet"/> with parameters available in <paramref name="template"/>.</returns>
         IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environmentSettings, ITemplate template);
```

Comment:
Should we modify this method?
`IsRequired` is now determined by `TemplatePriority`. Should it be adjusted given the enable\required is now conditioned?

How the conditions are displayed in help?